### PR TITLE
CEXT-2009 Add New Relic support to aio CLI

### PR DESCRIPTION
**Pull request by gjmarques**
## Description
Add New Relic support to log forwarding

## Related Issue
[Log forwarding New Relic support](https://github.com/adobe/aio-lib-runtime/issues/147)
[Add New Relic support to aio CLI](https://github.com/adobe/aio-cli-plugin-runtime/pull/311)

## Motivation and Context
Log Forwarding currently supports Splunk and Azure Log Analytics. This PR adds support for New Relic.

## How Has This Been Tested?
Auto tests are included

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by pablomoreno61 — APPROVED** [review]

**Review by aminakhyat — APPROVED** [review]

File: src/LogForwarding.js [hunk 1]
```diff
@@ -93,6 +93,23 @@ class LogForwarding {
     })
   }
 
+  /**
+   * Set Log Forwarding to New Relic
+   *
+   * @deprecated use `setDestination('new_relic', {...})`
```
Comment (buskamuza) [inline src/LogForwarding.js:99]: We shouldn't need this method. Old ones exist just to make sure we don't break backwards compatibility. `setDestination()` should be able to handle everything (used here https://github.com/adobe/aio-cli-plugin-runtime/pull/311/files#diff-243890719f28cb0b8642575ca636169f057c8f1d9773a9bc037d62f6b1ff8f93R21). Is this method used anywhere?

File: test/LogForwarding.test.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ const dataFixtures = [
     port: 'port1',
     index: 'index1',
     hec_token: 'token1'
+  }],
+  ['new_relic', 'setNewRelic', {
```
Comment (buskamuza) [inline test/LogForwarding.test.js:23]: This test is not necessary (as well as the method it covers). Those are written for deprecated methods. Maybe we even can get rid of those now?

File: test/LogForwardingUnderscoreNamespace.test.js [hunk 3]
```diff
@@ -18,6 +18,10 @@ const dataFixtures = [
     port: 'port1',
     index: 'index1',
     hec_token: 'token1'
+  }],
+  ['new_relic', 'setNewRelic', {
```
Comment (buskamuza) [inline test/LogForwardingUnderscoreNamespace.test.js:22]: Same for this case.

**Review by buskamuza — CHANGES_REQUESTED** [review]

File: src/LogForwarding.js (hunk 1 above)
Comment (gjmarques) [reply src/LogForwarding.js:99]: I think the method is used in these files:

- https://github.com/adobe/aio-cli-plugin-app/pull/689/files#diff-467e10a49ff9df960441ed6bf8f4910450aca0b8a28c29f9824cf64311d76584

- https://github.com/adobe/aio-cli-plugin-runtime/pull/311/files#diff-467e10a49ff9df960441ed6bf8f4910450aca0b8a28c29f9824cf64311d76584

Should I remove the `setNewRelic()` method then?

Comment (buskamuza) [reply src/LogForwarding.js:99]: I don't think it is necessary there. Those are just mocks for tests. Yes. Please, remove this method.

**Comment by shazron** [comment]
@gjmarques:
1. The comments above need to be resolved.
2. The unit tests are failing, can you please check? Thanks!

**Comment by gjmarques** [comment]
@shazron 
Comments above have been resolved and tests seem to be working locally

**Comment by shazron** [comment]
@gjmarques eslint failure (see failing test). Should be caught by `npm test` I think

**Comment by gjmarques** [comment]
@shazron 
Linting errors should have been fixed

**Review by shazron — APPROVED** [review]

**Review by buskamuza — APPROVED** [review]

**Review by buskamuza — APPROVED** [review]

**Pull request merged**